src/main/java/cinema/dao/impl/OrderDaoImpl.java

```diff
@@ -0,0 +1,53 @@
+package cinema.dao.impl;
+
+import cinema.dao.OrderDao;
+import cinema.exception.DataProcessingException;
+import cinema.injections.Dao;
+import cinema.model.Order;
+import cinema.model.User;
+import cinema.util.HibernateUtil;
+import java.util.List;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+
+@Dao
+public class OrderDaoImpl implements OrderDao {
+    @Override
+    public Order add(Order order) {
+        Transaction transaction = null;
+        Session session = null;
+        try {
+            session = HibernateUtil.getSessionFactory().openSession();
+            transaction = session.beginTransaction();
+            session.persist(order);
+            transaction.commit();
+            return order;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Could not insert " + order, e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public List<Order> getOrdersHistory(User user) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            return session.createQuery("select o from Order o "
+                    + "left join fetch o.user "
+                    + "left join fetch o.tickets t "
+                    + "left join fetch t.movieSession ms "
+                    + "left join fetch ms.movie "
+                    + "left join fetch ms.cinemaHall "
+                    + "where o.user = :user", Order.class)
+                    .setParameter("user", user)
+                    .getResultList();
```

Comment:
https://mateacademy.slack.com/archives/G01EHB8CWAE/p1612455330236200
check this case